### PR TITLE
Company Settings and Notifications

### PR DESCRIPTION
**Pull request by xXAndrexPereXx**
This PR was created to add `preferences` to the `Person` interface to configure the channels to send messages to the user (SMS, push email), create the `getCompanyData` function to get the company information of the authenticated user and create the `getAuthenticatedUserData` function to get the information of the logged in user.

### CODE REVIEW

File: src/clients/FlexbaseClient.Company.ts [hunk 1]
```diff
@@ -49,4 +67,31 @@ export class FlexbaseClientCompany extends FlexbaseClientBase {
             return { success: false, error: 'Unable to get company payments', payments: [] };
         }
     }
+
+    async getCompanyData(): Promise<CompanyResponse> {
+        try {
+            const response = await this.client.url(`/tenant`).get().json<ResponseCompanyData>();
+
+            return {
```
Comment (Mariana0711) [inline src/clients/FlexbaseClient.Company.ts:78]: We should check here the success of the response so based on that we could return something 

```
if (!response.success) {
       this.logger.error('While trying to get a banking deposit account, an unhandled exception was thrown', 
        response.error);
     }

    return { //... };
```

File: src/clients/FlexbaseClient.Person.ts [hunk 2]
```diff
@@ -134,4 +154,25 @@ export class FlexbaseClientPerson extends FlexbaseClientBase {
             return null;
         }
     }
+
+    async getAuthenticatedUserData(): Promise<PersonResponse> {
+        try {
+            const result =  await this.client.url('/user/self').get().json<Person>();
+            return {
```
Comment (Mariana0711) [inline src/clients/FlexbaseClient.Person.ts:164]: Same here

File: src/clients/FlexbaseClient.Company.ts (hunk 1 above)
Comment (xXAndrexPereXx) [reply src/clients/FlexbaseClient.Company.ts:78]: This endpoint doesn't return a `success`

File: src/clients/FlexbaseClient.Person.ts (hunk 2 above)
Comment (xXAndrexPereXx) [reply src/clients/FlexbaseClient.Person.ts:164]: This endpoint doesn't return a `success`

**Review by Mariana0711 — DISMISSED** [review]

**Review by huy-flex — DISMISSED** [review]

**Review by alegoos — APPROVED** [review]

**Review by huy-flex — APPROVED** [review]

**Pull request merged**